src/drivers/github.go

```diff
@@ -24,3 +35,125 @@ func (driver GithubCodeHostingDriver) GetNewPullRequestURL(repository string, br
 func (driver GithubCodeHostingDriver) GetRepositoryURL(repository string) string {
 	return "https://github.com/" + repository
 }
+
+// MergePullRequest merges the pull request through the Github API
+func (driver GithubCodeHostingDriver) MergePullRequest(options MergePullRequestOptions) error {
+	childPullRequestNumbers, err := driver.getPullRequestNumbersAgainst(options.Repository, options.Branch)
+	if err != nil {
+		return err
+	}
+	for _, childPullRequestNumber := range childPullRequestNumbers {
+		err = driver.updatePullRequestBase(options.Repository, childPullRequestNumber, options.ParentBranch)
+		if err != nil {
+			return err
+		}
+	}
+	pullRequestNumber, err := driver.getPullRequestNumberFor(options.Repository, options.Branch, options.ParentBranch)
+	if err != nil {
+		return err
+	}
+	query := url.Values{}
+	query.Add("access_token", os.Getenv("GITHUB_TOKEN"))
+	query.Add("commit_title", options.CommitTitle)
+	query.Add("commit_message", options.CommitMessage)
+	query.Add("sha", options.Sha)
+	query.Add("merge_method", options.MergeMethod)
+	mergePullRequestURL := fmt.Sprintf("https://api.github.com/repos/%s/pulls/%d/merge?%s", options.Repository, pullRequestNumber, query.Encode())
+	client := &http.Client{}
```

Comment:
I would extract talking to the Github API into a dedicated package. Before we do that, let's see if we can use https://github.com/google/go-github for talking to the Github API here. It handles all the Auth and rate limiting stuff already, is actively developed, and already tested.